src/PIL/DdsImagePlugin.py

```diff
@@ -124,114 +353,169 @@ def _open(self):
                 (dxgi_format,) = struct.unpack("<I", self.fp.read(4))
                 self.fp.read(16)
-                if dxgi_format in (DXGI_FORMAT_BC5_TYPELESS, DXGI_FORMAT_BC5_UNORM):
+                if dxgi_format in (
+                    DXGI_FORMAT.BC1_UNORM,
+                    DXGI_FORMAT.BC1_UNORM_SRGB,
+                    DXGI_FORMAT.BC1_TYPELESS,
+                ):
+                    self._mode = "RGBA"
+                    self.pixel_format = "BC1"
+                    tile = Image._Tile(
+                        "bcn", extents, data_offs, (1, self.pixel_format)
+                    )
+                elif dxgi_format in (DXGI_FORMAT.BC5_TYPELESS, DXGI_FORMAT.BC5_UNORM):
+                    self._mode = "RGB"
                     self.pixel_format = "BC5"
-                    n = 5
+                    tile = Image._Tile(
+                        "bcn", extents, data_offs, (5, self.pixel_format)
+                    )
+                elif dxgi_format == DXGI_FORMAT.BC5_SNORM:
                     self._mode = "RGB"
-                elif dxgi_format == DXGI_FORMAT_BC5_SNORM:
                     self.pixel_format = "BC5S"
-                    n = 5
+                    tile = Image._Tile(
+                        "bcn", extents, data_offs, (5, self.pixel_format)
+                    )
+                elif dxgi_format == DXGI_FORMAT.BC6H_UF16:
                     self._mode = "RGB"
-                elif dxgi_format == DXGI_FORMAT_BC6H_UF16:
                     self.pixel_format = "BC6H"
-                    n = 6
+                    tile = Image._Tile(
+                        "bcn", extents, data_offs, (6, self.pixel_format)
+                    )
+                elif dxgi_format == DXGI_FORMAT.BC6H_SF16:
                     self._mode = "RGB"
-                elif dxgi_format == DXGI_FORMAT_BC6H_SF16:
                     self.pixel_format = "BC6HS"
-                    n = 6
-                    self._mode = "RGB"
-                elif dxgi_format in (DXGI_FORMAT_BC7_TYPELESS, DXGI_FORMAT_BC7_UNORM):
+                    tile = Image._Tile(
+                        "bcn", extents, data_offs, (6, self.pixel_format)
+                    )
+                elif dxgi_format in (DXGI_FORMAT.BC7_TYPELESS, DXGI_FORMAT.BC7_UNORM):
+                    self._mode = "RGBA"
                     self.pixel_format = "BC7"
-                    n = 7
-                elif dxgi_format == DXGI_FORMAT_BC7_UNORM_SRGB:
+                    tile = Image._Tile(
+                        "bcn", extents, data_offs, (7, self.pixel_format)
+                    )
+                elif dxgi_format == DXGI_FORMAT.BC7_UNORM_SRGB:
+                    self._mode = "RGBA"
                     self.pixel_format = "BC7"
                     self.info["gamma"] = 1 / 2.2
-                    n = 7
+                    tile = Image._Tile(
+                        "bcn", extents, data_offs, (7, self.pixel_format)
+                    )
```

Comment:
What is the thinking behind this change, where instead of specifying different `n` values, you write out the entire tile each time? It would seem to be increasing the amount of duplicate code.

Reply:
Hm, yeah, you're right. I will refactor it

Reply:
After second though, the way i structured if-elif chain does not allow me to easily refactor it back to n= thing

Reply:
See what you think of https://github.com/REDxEYE/Pillow/pull/6